src/CamelKJSONUtils.ts

```diff
@@ -0,0 +1,147 @@
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as fs from 'fs';
+import * as vscode from 'vscode';
+import * as http from "http";
+
+export const proxyURLSetting = 'camelk.integrations.proxyURL';
+export const proxyNamespaceSetting = 'camelk.integrations.proxyNamespace';
+
+const BACKSLASH = '\u005C';
+
+export function createCamelKRestURL() : string {
+	let server = vscode.workspace.getConfiguration().get(proxyURLSetting);
+	let namespace = vscode.workspace.getConfiguration().get(proxyNamespaceSetting);
+	let outputUrl = server + "/apis/camel.apache.org/v1alpha1/namespaces/" + namespace + "/integrations";
+	return outputUrl;
+}
+
+export function createCamelKDeleteRestURL(integrationName:string) : string {
+	let baseUrl = createCamelKRestURL();
+	let outputUrl = baseUrl + "/" + integrationName;
+	return outputUrl;
+}
+
+export function createCamelKPodLogURL(podName:string) : string {
+	let server = vscode.workspace.getConfiguration().get(proxyURLSetting);
+	let namespace = vscode.workspace.getConfiguration().get(proxyNamespaceSetting);
+	let outputUrl = server + "/api/v1/namespaces/" + namespace + "/pods/" + podName + "/log";
+	return outputUrl;
+}
+
+export function createCamelKGetPodsURL() : string {
+	let server = vscode.workspace.getConfiguration().get(proxyURLSetting);
+	let namespace = vscode.workspace.getConfiguration().get(proxyNamespaceSetting);
+	let outputUrl = server + "/api/v1/namespaces/" + namespace + "/pods";
+	return outputUrl;
+}
+
+function cleanText(text:string) : Promise<string> {
+	return new Promise( (resolve, reject) => {
+		if (text) {
+			let reg = new RegExp('(?:\\\")*');
+			let cleanedup = text.replace(reg, '');
+			var noCR = cleanedup.replace(/\\r/g, BACKSLASH + 'r');
+			var noLF = noCR.replace(/\\n/g, BACKSLASH + 'n');
+			var noT = noLF.replace(/\\t/g, BACKSLASH + 't');
+			let noDblSlash = noT.replace(/\\/g, BACKSLASH);
+			resolve(noDblSlash);
+		} else {
+			reject();
+		}
+	});
+}
+
+export function stringifyFileContents(absoluteFilePath:string) : Promise<string> {
+	return new Promise( (resolve, reject) => {
+		var text = fs.readFileSync(absoluteFilePath);
+		if (text) {
+			let textStr = text.toString();
+			cleanText(textStr)
+				.then( (cleanedText) => resolve(cleanedText))
+				.catch ( () => reject());
+		} else {
+			reject();
+		}
+	});
+}
+
+export function createCamelKDeployJSON( name:string, fileContent:string, fileName:string) : Promise<string> {
+	return new Promise( (resolve) => {
+		let content = {
+			"kind":"Integration",
+			"apiVersion":"camel.apache.org/v1alpha1",
```

Comment:
Great question. I have asked the powers that be about the future of this version.

Reply:
In the meantime, I'll spin off the "v1alpha1" as a constant so I can change it one place when it does change.

Reply:
The API will change eventually to v1 instead of v1alpha1, but there will be a transition period per Nicola. The API is here - https://github.com/apache/camel-k/tree/master/pkg/apis/camel